### ISSUE TITLE
namespace "thrust" has no member "device"

### ISSUE THREAD

**Issue by ssvenkatesh** [NVIDIA/thrust#904]
Code:
[search.cu.txt](https://github.com/thrust/thrust/files/1649543/search.cu.txt)

Getting below compilation error
Search.cu(28): error : namespace "thrust" has no member "device"
on compiling 
	iter = thrust::find(thrust::device, d_vec.begin(), d_vec.end(), 200193); 

"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.1\bin\nvcc.exe" -gencode=arch=compute_30,code=\"sm_30,compute_30\" --use-local-env --cl-version 2017 -ccbin "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.11.25503\bin\HostX86\x64" -x cu  -I.\ -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.1\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.1\include"     --keep-dir x64\Release -maxrregcount=0  --machine 64 --compile -cudart static     -DWIN32 -DWIN64 -DNDEBUG -D_CONSOLE -D_MBCS -Xcompiler "/EHsc /W3 /nologo /Od /FS /Zi  /MD " -o x64\Release\Search.cu.obj "Search.cu"  

OS: Windows 8.1
GPU: NVIDIA Quadro K2200
Visual Studio : 2017
CUDA: 9.1
Thrust Version: 1.8.2

But the Same code works with visual studio 2013 , CUDA 9.0

Am I missing something here.?

**Comment by brycelelbach** [comment]
Hi,

This report does not have enough information to be actionable. Please read the guidelines [here](https://github.com/brycelelbach/cpp_bug_reporting_guidelines) and provide an updated report.

**Comment by ssvenkatesh** [comment]
Updates Code, Environment and GPU info.

**Comment by nimerix** [comment]
You need to include the execution policy header for `thrust::device`
```cpp
#include <thrust/execution_policy.h>
```
See [thrust/find.h@81](https://github.com/thrust/thrust/blob/8551c97870cd722486ba7834ae9d867f13e299ad/thrust/find.h#L81)

**Comment by brycelelbach** [comment]
Hey ssvenkatesh, have you tried nimerix's suggestion?
